packages/table/src/presenters/TablePresenter.js

```diff
@@ -266,222 +283,251 @@ const TablePresenter = ({
-                              onTableCellClick={onTableCellClick}
-                              getActiveMultiSelectRowArray={
-                                getActiveMultiSelectRowArray
-                              }
-                              setAllMultiSelectedRows={setAllMultiSelectedRows}
-                              setActiveMultiSelectRowArray={
-                                setActiveMultiSelectRowArray
-                              }
-                              rowTypeToMap={paginateDynamic ? rows : page}
-                              customStylesheet={customStylesheet}
-                            >
-                              {/* eslint-disable */}
-                              {cell.isGrouped ? (
-                                <>
-                                  <span {...row.getToggleRowExpandedProps()}>
-                                    <GroupElements isExpanded={row.isExpanded}>
-                                      {meta.groupElements}
-                                    </GroupElements>
-                                  </span>{" "}
-                                  {renderCellData(meta.formatDate, cell)} ({row.subRows.length})
-                                </>
-                              ) : cell.isAggregated ? (
-                                cell.render("Aggregated")
-                              ) : cell.isPlaceholder ? null : (
-                                renderCellData(meta.formatDate, cell)
-                              )}
-                            </TableDataCell>
-                          );
-                        })}
-                      </div>
-                      {meta.expandedComponent && (
-                        <ExpandedContent
-                          curItem={customContentArray[rowIndex]}
-                          expandedContentStyles={rowStyles}
-                        >
-                          {meta.expandedComponent}
-                        </ExpandedContent>
-                      )}
+                              onClick={event => event.stopPropagation()}
+                            />
+                            {meta.sortColumns && column.id !== "selection" ? (
+                              <SortColumns
+                                isSorted={column.isSorted}
+                                isSortedDesc={column.isSortedDesc}
+                                density={metadata.densityId}
+                              >
+                                {meta.sortColumns}
+                              </SortColumns>
+                            ) : (
+                              ""
+                            )}
+                          </TableHeaderCell>
+                        );
+                      })}
                     </div>
-                  );
-                })}
+                  ))}
               </div>
+              {isGrouped && (
+                <Accordion
```

Comment:
wondering if it would might be more future proof if we went `isGrouped && UserGroupingComponent (<UserGroupingComponent />)` that way down the line if other folks have other grouping designs they can use the components of their choice?

Reply:
updated to add DataGroupComponent to future proof the grouping of data items.